### PR TITLE
Add async library support

### PR DESCRIPTION
**Pull request by Paillat-dev**
Add async support on the entire library -- simply use `AsyncClient` instead of `Client` if you want to work in an asyncronous environment.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 main.py
 rewrite.py
 discordoauth2/__pycache__
+.idea/
```
Comment (treeben77) [inline .gitignore:4]: I don't believe `.idea/` needs to be here.

File: discordoauth2/__init__.py [hunk 2]
```diff
@@ -1,296 +1,12 @@
...
-        pass
-
-    class RateLimited(HTTPException):
-        def __init__(self, text, retry_after):
-            self.retry_after = retry_after
-            super().__init__(text)
-  
-    class Forbidden(HTTPException):
-        pass
+from .async_oauth import AsyncClient, AsyncAccessToken, AsyncPartialAccessToken
```
Comment (treeben77) [inline discordoauth2/__init__.py:2]: why don't we import exceptions from the async version, wouldn't this cause issues with error handling with aysnc?

**Review by treeben77 — CHANGES_REQUESTED** [review]
I approve of the idea of adding async support! However, there are a few comments I left on your changes that I believe need to be addressed first.

File: .gitignore (hunk 1 above)
Comment (LawMixer) [reply .gitignore:4]: its another IDE files -It's  intellij IDE.

File: discordoauth2/__init__.py (hunk 2 above)
Comment (LawMixer) [reply discordoauth2/__init__.py:2]: we should separate the exceptions to a different file for ease of organization;

File: .gitignore (hunk 1 above)
Comment (Paillat-dev) [reply .gitignore:4]: Yeah if you use any of the jetbrains ides

File: discordoauth2/__init__.py (hunk 2 above)
Comment (Paillat-dev) [reply discordoauth2/__init__.py:2]: I actually wanted to add that to the pr then though I should stop working for the day went to sleep then didn't sleep and saw my emails. Il will change that tomorrow.

Comment (Paillat-dev) [reply discordoauth2/__init__.py:2]: Cuz I saw how bad I did that part but was too tired to work on it

**Comment by Paillat-dev** [comment]
Here. I did not remove `.idea/` from the `.gitignore`, it is from the JetBrains IDEs.

**Review by treeben77 — APPROVED** [review]
Everything looks good to me now! I'll let @LawMixer review it all too.

**Review by LawMixer — APPROVED** [review]
looks good to me, not unless @treeben77 wants to have a seperate pr for the formatting changes and all of that

**Review by treeben77 — APPROVED** [review]

**Review by LawMixer — APPROVED** [review]

**Pull request merged**